server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1107,34 +1097,11 @@ private WildcardExpressionResolver() {
             // Utility class
         }
 
-        /**
-         * Returns a collection of resource names given the {@param expressions} which contains wildcards and exclusions.
-         */
-        public static Collection<String> resolve(Context context, List<String> expressions) {
-            Objects.requireNonNull(expressions);
-            if (context.getOptions().expandWildcardExpressions() == false) {
-                if (expressions.size() == 1 && expressions.get(0).equals(Metadata.ALL)) {
-                    return List.of();
-                } else {
-                    if (context.getOptions().ignoreUnavailable() == false) {
-                        for (String expression : expressions) {
-                            ensureAliasOrIndexExists(context, expression);
-                        }
-                    }
-                    return expressions;
-                }
-            } else if (isEmptyOrTrivialWildcard(expressions)) {
-                return innerResolveAll(context);
-            } else {
-                return innerResolve(context, expressions);
-            }
-        }
-
         /**
          * Returns all the indices and all the datastreams, considering the open/closed, system, and hidden context parameters.
          * Depending on the context, returns the names of the datastreams themselves or their backing indices.
          */
-        private static Collection<String> innerResolveAll(Context context) {
+        public static Collection<String> resolveAll(Context context) {
```

Comment:
This is now exposed because the `_all` expression is handled separately, rather than assume that `_all` can appear in any expression at any point.